### ISSUE TITLE
GM_download onprogress stopped working in chrome

### ISSUE THREAD

**Issue by martixy** [Tampermonkey/tampermonkey#2231]
## Expected Behavior
Callback is triggered whenever you get more data.

## Actual Behavior
Nothing happens anymore.
This is recent, since maybe last week or two.
This is an issue only on chrome, works on FF. Might have been a recent chrome update. Seriously fuck MV3.

## Specifications

- Chromium: 130.0.6723.92
- TM: v5.3.1
- OS: Win10 22H2

## Script

```js
GM_download({
    url: "https://i.imgur.com/lYuZ60P.png",
    name: "kazumaDesu.jpg",
    onprogress(progress) {
        console.log(progress)
    }
})
```

**Comment by zhuozhiyongde** [comment]
For me, Chrome works fine, but Arc based on Chromium does not work properly.

**Comment by BorisChuev** [comment]
Sent with OfficeSuite Mail for Windows PC
------------- Original Message --------------From: Arthals ***@***.***>Sent: 10.11.2024 08:56:38To: Tampermonkey/tampermonkey ***@***.***>Cc: Subscribed ***@***.***>Subject: Re: [Tampermonkey/tampermonkey] GM_download onprogress stopped working in chrome (Issue #2231)




For me, Chrome works fine, but Arc based on Chromium does not work properly.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

--=-uBnuzeFoDEeTCURyaCkvTg==
Content-Type: text/html; charset=utf-8
Content-Id: ***@***.***>
Content-Transfer-Encoding: quoted-printable

<html><head><style id=3D"_osm_style"></style></head><body><div><br></div><div><br></div><div data-signature-c=
ontainer=3D""><div data-signature=3D"true"><div>Sent with <a href=3D"https:=
//www.officesuite.com/?referrer=3Dmail&amp;utm_source=3Dbranded-mail&amp;ut=
m_medium=3Doswin-mail&amp;utm_campaign=3Dexternal-traffic-oswin-branded-mai=
l-link&amp;utm_content=3Demail-w-oswin-link" style=3D"text-decoration:none;=
 color:#2562AB;">OfficeSuite</a> Mail for Windows PC</div></div><div data-s=
ignature-margin-bottom=3D"true"><br></div></div><div id=3D"original-message=
-content"><div><div id=3D"_osm-template-type">------------- Original Messag=
e --------------</div><div><div><strong>From:</strong> <span id=3D"_osm-rep=
ly-from">Arthals ***@***.***&gt;</span></div><div><strong>=
Sent:</strong> <span id=3D"_osm-reply-sent">10.11.2024 08:56:38</span></div=
><div><strong>To:</strong> <span id=3D"_osm-reply-to">Tampermonkey/tampermo=
nkey ***@***.***&gt;</span></div><div><strong>Cc:</=
strong> <span id=3D"_osm-reply-cc">Subscribed ***@***.***=
.com&gt;</span></div><div><strong>Subject:</strong> <span id=3D"_osm-reply-=
subject">Re: [Tampermonkey/tampermonkey] GM_download onprogress stopped wor=
king in chrome (Issue #2231)</span></div></div></div><br><blockquote style=3D=
"border:none;border-left:solid #CCCCCC 1.0pt;padding:0cm 0cm 0cm 8.0pt;marg=
in-left:3.75pt;margin-top:2.25pt;margin-right:0cm;margin-bottom:5.0pt;"><di=
v><p><br></p><p dir=3D"auto">For me, Chrome works fine, but Arc based on Ch=
romium does not work properly.</p><p style=3D"font-size:small;-webkit-text-=
size-adjust:none;color:#666;">=E2=80=94<br>Reply to this email directly, <a=
 href=3D"https://github.com/Tampermonkey/tampermonkey/issues/2231#issuecomm=
ent-2466599327">view it on GitHub</a>, or <a href=3D"https://github.com/not=
ifications/unsubscribe-auth/A7K4GC2LHF6V7YC3TU4DVULZ73YRNAVCNFSM6AAAAABRLU3=
GEWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRWGU4TSMZSG4">unsubscribe</a=
>.<br>You are receiving this because you are subscribed to this thread.<img=
 src=3D"https://github.com/notifications/beacon/A7K4GC6PDTG7OBEJ7P7PU6TZ73Y=
RNA5CNFSM6AAAAABRLU3GEWWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3=
TUL5UWJTUTAVIZ6.gif" height=3D"1" width=3D"1" alt=3D""><span style=3D"color=
: transparent; font-size: 0; display: none; visibility: hidden; overflow: h=
idden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hi=
de: all;">Message ID: <span>&lt;Tampermonkey/tampermonkey/issues/2231/24665=
99327</span><span>@</span><span>github</span><span>.</span><span>com&gt;</s=
pan></span></p></div></blockquote></div></body></html>=

--=-uBnuzeFoDEeTCURyaCkvTg==--

**Comment by derjanb** [comment]
Please go to "chrome://extensions", enable developer mode and press the "Update" button. Version 5.3.2 should fix this.

**Comment by zhuozhiyongde** [comment]
> Please go to "chrome://extensions", enable developer mode and press the "Update" button. Version 5.3.2 should fix this.

It still not work for me.

**Comment by martixy** [comment]
@derjanb 5.3.2 does not fix the issue.

**Comment by zhuozhiyongde** [comment]
Update:
Chrome is now unavailable.
- Chrome Version: 131.0.6778.140
- TamperMonkey Version: 5.3.3

**Comment by zhuozhiyongde** [comment]
After switching to use the browser's native API for downloading in the settings, the issue was resolved.

**Comment by martixy** [comment]
Well... it's an okay workaround. But the behaviour is different:
1. It asks you for a save location when starting the download, rather than when completing it.
2. It updates much less frequently.
3. It doesn't update at the end of the download. (Meaning in this case you have to take the values from previous onprogress callbacks and pass them to onload, if you need that data.)

**Comment by zhuozhiyongde** [comment]
> Well... it's an okay workaround. But the behaviour is different:
> 1. It asks you for a save location when starting the download, rather than when completing it.
> 2. It updates much less frequently.
> 3. It doesn't update at the end of the download. (Meaning in this case you have to take the values from previous onprogress callbacks and pass them to onload, if you need that data.)

You are right,but in my case, I only need to rename the download file, so it works :)

**Comment by martixy** [comment]
The above are annoying at best.

But browser API mode has one actually serious usability problem - it never remembers the download location, always taking you back to the one set in the settings.

**Comment by derjanb** [comment]
This should be fixed at 5.4.6226 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi - still in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` **once the review is finished**.

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by zhuozhiyongde** [comment]
Thx!

**Comment by martixy** [comment]
Can confirm, fixed in beta 5.4.6226. Tnx for fix. Was an annoying bug.

I assume issue should remain open until it's in stable.

**Comment by zhuozhiyongde** [comment]
In my case, the download still stops at 99% and cannot complete, but there is indeed an `onprogress` callback? Is there something wrong with my code?

https://github.com/zhuozhiyongde/PKU-Art/blob/9847b978f4604626d4eb099a7908601f6e28ffcb/src/main.js#L588-L635

**Comment by fushengruomengzhang** [comment]
> This should be fixed at 5.4.6226 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi - still in review)
> 
> Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`. Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` **once the review is finished**.
> 
> For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

Lack of response
